lib/open_api_spex/open_api/decode.ex

```diff
@@ -403,4 +404,13 @@ defmodule OpenApiSpex.OpenApi.Decode do
       to_struct(v, mod)
     end)
   end
+
+  defp manage_additional_properties(%_{additionalProperties: %{"$ref" => reference}} = map) do
+    Map.put(map, :additionalProperties, %Reference{"$ref": reference})
```

Comment:
This is fine, but it's good practice to use map literal syntax to put key values on a struct, like this:

```
%{schema | additionalProperties: %Reference{"$ref": reference}}
```

That way, if there is a typo in `additionalProperties:`, the runtime will raise an exception.

Plus, the syntax is shorter.

Reply:
I code in elixir since some months now and never realized this! Super thanks!